test/unit_CartesianProduct.jl

```diff
@@ -141,6 +145,13 @@ for N in [Float64, Float32, Rational{Int}]
     H = convert(Hyperrectangle, b1 × h2)
     @test low(H) == N[0, 2, 4] && high(H) == N[1, 3, 5]
 
+    # convert a hyperrectangle to a cartesian product of intervals
+    H = Hyperrectangle(N[0, 0], N[1, 1])
+    Hcp = convert(CartesianProduct{N, Interval{N}, Interval{N}}, H)
+    @test Hcp isa CartesianProduct &&
+          Hcp.X == Interval(N(-1), N(1)) &&
+          Hcp.Y == Interval(N(-1), N(1))
```

Comment:
Can we have different entries to test that the order is correct?
`H = Hyperrectangle(N[0, 1], N[1, 3])` and then `Hcp.X == Interval(N(-1), N(1)) && Hcp.Y == Interval(N(-2), N(4))`